app/controllers/users_controller.rb

```diff
@@ -77,6 +79,30 @@ def change_password
     redirect_to edit_user_path unless current_user.greenlight_account?
   end
 
+  # GET /u/:user_uid/join_settings
+  def join_settings
+    redirect_to root_path unless current_user
+  end
+
+  # POST /u/:user_uid/join_settings
+  def update_settings
+    return redirect_to root_path unless current_user
+    settings_params = params.require(:user).permit(@join_settings)
+    @user.update_all_user_settings(settings_params)
+
+    # Notify the user that their account has been updated.
+    return redirect_back fallback_location: root_path, flash: { success: I18n.t("info_update_success") } if
+      @user.save
+
+    # redirect_to change_password_path
```

Comment:
Invalid comment

Reply:
I have make changes for this please review it